sympy/functions/elementary/exponential.py

```diff
@@ -456,6 +456,19 @@ def _eval_rewrite_as_tanh(self, arg):
         from sympy import tanh
         return (1 + tanh(arg/2))/(1 - tanh(arg/2))
 
+    def _eval_rewrite_as_sqrt(self, arg):
+        from sympy.functions.elementary.trigonometric import sin, cos
+        if arg.is_Mul:
+            coeff = arg.coeff(S.Pi*S.ImaginaryUnit)
+            if coeff and coeff.is_number:
+                cosine, sine = cos(S.Pi*coeff), sin(S.Pi*coeff)
+                if not isinstance(cosine, cos) and not isinstance (sine, sin):
+                    return cosine + S.ImaginaryUnit*sine
+            else:
```

Comment:
This would probably work without these four last lines. I think the code looks better without them.

Reply:
@jksuom but then what will be returned when `if coeff and coeff.is_number` becomes true?
I mean cos and sin will produce sqrt but after removing cos and sin how will it work?

Reply:
I gave input `exp(pi*I/4).rewrite(sqrt)` after removing last 4 lines and the output that I got is `exp(I*pi/4)`

Reply:
I mean the four lines *after* this one, lines 467 - 470.

Reply:
If `coeff and coeff.is_number` is true, then the rewritten form will be returned. Otherwise `None` will be returned, and then the caller should return the original.

Reply:
@jksuom removed the last 4 lines